packages/pubsub/src/connection-pool.js

```diff
@@ -459,4 +477,65 @@ ConnectionPool.prototype.shouldReconnect = function(status) {
   return true;
 };
 
+/**
+ *
+ */
+ConnectionPool.prototype.watchForReady = function() {
```

Comment:
The pattern of `watching` then `waiting` "forReady" is a bit confusing. Can these two methods be merged into one, where it's just "waitForReady()"?

Reply:
Sure! Done.